yew/src/virtual_dom/mod.rs

```diff
@@ -315,109 +315,120 @@ impl Default for Attributes {
+        self.set
+            .extend(iter.into_iter().map(|x| x.as_ref().to_string().into()));
+    }
+}
 
-    /// Adds other classes to this set of classes; returning itself.
-    ///
-    /// Takes the logical union of both `Classes`.
-    pub fn extend<T: Into<Classes>>(mut self, other: T) -> Self {
-        self.set.extend(other.into().set.into_iter());
-        self
+impl IntoIterator for Classes {
+    type Item = Cow<'static, str>;
+    type IntoIter = indexmap::set::IntoIter<Cow<'static, str>>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.set.into_iter()
     }
 }
 
 impl ToString for Classes {
     fn to_string(&self) -> String {
         self.set
             .iter()
-            .map(String::as_str)
-            .collect::<Vec<&str>>()
+            .map(Borrow::borrow)
+            .collect::<Vec<_>>()
             .join(" ")
     }
 }
 
-impl From<&str> for Classes {
-    fn from(t: &str) -> Self {
+impl From<&'static str> for Classes {
+    fn from(t: &'static str) -> Self {
         let set = t
             .split_whitespace()
-            .map(String::from)
             .filter(|c| !c.is_empty())
+            .map(|x| x.into())
             .collect();
         Self { set }
     }
 }
 
 impl From<String> for Classes {
     fn from(t: String) -> Self {
-        Classes::from(t.as_str())
+        let mut set = IndexSet::new();
+        set.insert(t.into());
+        Self { set }
     }
 }
 
 impl From<&String> for Classes {
     fn from(t: &String) -> Self {
-        Classes::from(t.as_str())
+        Classes::from(t.clone())
     }
 }
 
-impl<T: AsRef<str>> From<Option<T>> for Classes {
+impl<T: Into<Classes>> From<Option<T>> for Classes {
     fn from(t: Option<T>) -> Self {
-        t.as_ref()
-            .map(|s| <Classes as From<&str>>::from(s.as_ref()))
-            .unwrap_or_default()
+        t.map(|x| x.into()).unwrap_or_default()
     }
 }
 
-impl<T: AsRef<str>> From<&Option<T>> for Classes {
+impl<T: Into<Classes> + Clone> From<&Option<T>> for Classes {
     fn from(t: &Option<T>) -> Self {
-        t.as_ref()
-            .map(|s| <Classes as From<&str>>::from(s.as_ref()))
-            .unwrap_or_default()
+        t.clone().into()
     }
 }
 
-impl<T: AsRef<str>> From<Vec<T>> for Classes {
+impl<T: Into<Classes> + Clone> From<Vec<T>> for Classes {
     fn from(t: Vec<T>) -> Self {
         Classes::from(t.as_slice())
     }
 }
 
-impl<T: AsRef<str>> From<&[T]> for Classes {
+impl<T: Into<Classes> + Clone> From<&[T]> for Classes {
     fn from(t: &[T]) -> Self {
-        let set = t
-            .iter()
-            .map(|x| x.as_ref())
-            .flat_map(|s| s.split_whitespace())
-            .map(String::from)
-            .filter(|c| !c.is_empty())
-            .collect();
-        Self { set }
+        let mut classes = Classes::with_capacity(t.len());
+        for class_ref in t {
+            classes.push(class_ref.clone().into());
+        }
+        classes
```

Comment:
Can be simplified to this:

```suggestion
        Self::from_iter(t.iter().cloned())
```

Btw, we could also use `.collect()` instead of `Self::from_iter`. I just chose to use the latter because it makes it easier  to locate the implementation.

Reply:
Thanks for the comment